card/src/main/java/com/adyen/checkout/card/CardDelegate.kt

```diff
@@ -8,27 +8,21 @@
 
 package com.adyen.checkout.card
 
-import com.adyen.checkout.card.api.PublicKeyConnection
 import com.adyen.checkout.card.data.CardType
 import com.adyen.checkout.card.data.DetectedCardType
 import com.adyen.checkout.card.data.ExpiryDate
-import com.adyen.checkout.components.api.suspendedCall
+import com.adyen.checkout.card.repository.PublicKeyRepository
 import com.adyen.checkout.components.base.PaymentMethodDelegate
 import com.adyen.checkout.components.validation.ValidatedField
-import com.adyen.checkout.core.log.LogUtil
-import com.adyen.checkout.core.log.Logger
 import kotlinx.coroutines.CoroutineScope
-import org.json.JSONException
-import java.io.IOException
-
-private val TAG = LogUtil.getTag()
-private const val CONNECTION_RETRIES = 3
 
 @Suppress("TooManyFunctions")
 abstract class CardDelegate(protected val cardConfiguration: CardConfiguration) : PaymentMethodDelegate {
 
     protected val noCvcBrands: Set<CardType> = hashSetOf(CardType.BCMC)
 
+    private val publicKeyRepository: PublicKeyRepository = PublicKeyRepository()
```

Comment:
should we pass the repository in the constructor to allow for manual DI and testing?